apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java

```diff
@@ -335,38 +376,44 @@ public static Stream<LoadedFileFilter> buildTrimmedFilters(
   /**
    * Build a filter for this loaded file. Should be a pure function.
    *
-   * @param fileId to build a filter for
-   * @param firstUpdated time stamp
+   * @param tuple the {@link LoadedTuple} for a given file
    * @param fetchById a function which returns a list of batches
+   * @param fetchBatchSizeByFileId a function that returns the batch size of the file
+   * @param fetchEstimatedBenesCountByFileId a function that returns an estimated count of
+   *     beneficiaries per-{@link LoadedBatch} for the given {@link LoadedTuple} file ID
    * @return a new filter
    */
   public static LoadedFileFilter buildFilter(
-      long fileId, Instant firstUpdated, Function<Long, List<LoadedBatch>> fetchById) {
-    final List<LoadedBatch> loadedBatches = fetchById.apply(fileId);
-    final int batchCount = loadedBatches.size();
+      LoadedTuple tuple,
+      BiFunction<Long, Integer, Stream<LoadedBatch>> fetchById,
+      LongFunction<Long> fetchBatchSizeByFileId,
+      LongFunction<Long> fetchEstimatedBenesCountByFileId) {
+    final var fileId = tuple.getLoadedFileId();
+    final var batchCount = fetchBatchSizeByFileId.apply(fileId).intValue();
     if (batchCount == 0) {
       throw new IllegalArgumentException("Batches cannot be empty for a filter");
     }
-    final String[] loadedBatchBenes = loadedBatches.get(0).getBeneficiaries().split(",");
-    final int batchSize = loadedBatchBenes.length;
 
-    // It is important to get a good estimate of the number of entries for
-    // an accurate FFP and minimal memory size. This one assumes that all batches are of equal size.
-    final BloomFilter bloomFilter = LoadedFileFilter.createFilter(batchSize * batchCount);
-
-    // Loop through all batches, filling the bloom filter and finding the lastUpdated
-    Instant lastUpdated = firstUpdated;
-    for (LoadedBatch batch : loadedBatches) {
-      for (Long beneficiary : batch.getBeneficiariesAsList()) {
-        bloomFilter.putLong(beneficiary);
-      }
-      if (batch.getCreated().isAfter(lastUpdated)) {
-        lastUpdated = batch.getCreated();
-      }
```

Comment:
The logic to track the `lastUpdated` was completely unnecessary. The query for the list of `LoadedTuple`s returns the `max(loaded_batches.created)` of the `loaded_batches` associated with the given `loaded_file_id`. So, each `LoadedTuple` _already has `lastUpdated`_ prior to calling this method.

I've verified that this is true by looking at the `lastUpdated`s of every `LoadedTuple` and `LoadedFileFilter` (built using this logic) ensuring they're equal.

I've opted to completely remove this, which simplifies this a bit.